src/statement-list/statement_list_parse_tree_io.cpp

```diff
@@ -0,0 +1,191 @@
+  out << "Version: " << module.version << "\n\n";
+}
+
+void output_return_value(
+  const statement_list_parse_treet::functiont &function,
+  std::ostream &out)
+{
+  if(function.return_value.is_nil())
+    out << "Return Value: Void \n\n";
+  else
+    out << "Return Value: " << function.return_value.id() << "\n\n";
+}
+
+void output_common_var_declarations(
+  std::ostream &out,
+  const statement_list_parse_treet::tia_modulet &module)
+{
+  if(!module.var_input.empty())
+  {
+    out << "--------- Input Variables ----------\n\n";
+    output_var_declaration_list(out, module.var_input);
+  }
+
+  if(!module.var_inout.empty())
+  {
+    out << "--------- In/Out Variables ---------\n\n";
+    output_var_declaration_list(out, module.var_inout);
+  }
+
+  if(!module.var_output.empty())
+  {
+    out << "--------- Output Variables ---------\n\n";
+    output_var_declaration_list(out, module.var_output);
+  }
+
+  if(!module.var_constant.empty())
+  {
+    out << "-------- Constant Variables --------\n\n";
+    output_var_declaration_list(out, module.var_constant);
+  }
+
+  if(!module.var_temp.empty())
+  {
+    out << "---------- Temp Variables ----------\n\n";
+    output_var_declaration_list(out, module.var_temp);
+  }
+}
+
+void output_static_var_declarations(
+  std::ostream &out,
+  const statement_list_parse_treet::function_blockt &block)
+{
+  if(!block.var_static.empty())
+  {
+    out << "--------- Static Variables ---------\n\n";
+    output_var_declaration_list(out, block.var_static);
+  }
+}
+
+void output_var_declaration_list(
+  std::ostream &out,
+  const statement_list_parse_treet::var_declarationst &declarations)
+{
+  for(const auto &declaration : declarations)
+  {
+    output_var_declaration(out, declaration);
+    out << "\n\n";
+  }
+}
+
+void output_var_declaration(
+  std::ostream &out,
+  const statement_list_parse_treet::var_declarationt &declaration)
+{
+  out << declaration.variable.pretty() << "\n";
+  if(declaration.default_value.has_value())
+    out << "  * default_value: "
+        << declaration.default_value.value().get(ID_value);
+  else
+    out << "  * default_value: (none)";
+}
+
+void output_network_list(
+  std::ostream &out,
+  const statement_list_parse_treet::networkst &networks)
+{
+  out << "-------------- Networks --------------\n\n";
+  for(const auto &network : networks)
+  {
+    output_network(out, network);
+    out << "\n";
+  }
+}
+
+void output_network(
+  std::ostream &out,
+  const statement_list_parse_treet::networkt &network)
+{
+  out << "Title: " << network.title.value_or("(none)") << "\n";
+  out << "Instructions: ";
+  if(network.instructions.empty())
+    out << "(none)";
+  out << "\n";
+  for(const auto &instruction : network.instructions)
+  {
+    output_instruction(out, instruction);
+    out << "\n";
+  }
+}
+
+void output_instruction(
+  std::ostream &out,
+  const statement_list_parse_treet::instructiont &instruction)
+{
+  for(const auto &token : instruction.tokens)
```

Comment:
Without knowing the type of `instruction.tokens`, it is not obvious to the reader what the type of `token` will be. Avoid `auto` in these contexts and use the actual type.